common/src/main/java/com/microsoft/identity/common/SharedPreferencesHelper.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.identity.common;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+/**
+ * Convenience class for accessing {@link SharedPreferences}.
+ */
+public class SharedPreferencesHelper {
+
+    private final SharedPreferences mSharedPreferences;
+
+    /**
+     * Constructs an instance of SharedPreferencesHelper.
+     * The default operating mode is {@link Context#MODE_PRIVATE}
+     *
+     * @param context Interface to global information about an application environment.
+     * @param name    The desired {@link android.content.SharedPreferences} file. It will be created
+     *                if it does not exist.
+     */
+    public SharedPreferencesHelper(
+            final Context context,
+            final String name) {
+        mSharedPreferences = context.getSharedPreferences(name, Context.MODE_PRIVATE);
+    }
+
+    /**
+     * Constructs an instance of SharedPreferencesHelper.
+     *
+     * @param context       Interface to global information about an application enviroment.
+     * @param name          The desired {@link SharedPreferences} file. It will be created
+     *                      if it does not exist.
+     * @param operatingMode Operating mode {@link Context#getSharedPreferences(String, int)}.
+     */
+    public SharedPreferencesHelper(
+            final Context context,
+            final String name,
+            final int operatingMode) {
+        mSharedPreferences = context.getSharedPreferences(name, operatingMode);
+    }
+
+    /**
+     * Saves a Token (as a {@link String} to the {@link SharedPreferences} file.
+     *
+     * @param key   The name (key) of the Token to save.
+     * @param value The Token's value (as a {@link String}).
+     * @return True, if the value was successfully written.
+     */
+    public final boolean saveToken(
+            final String key,
+            final String value) {
+        final SharedPreferences.Editor editor = mSharedPreferences.edit();
+        editor.putString(key, value);
+        return editor.commit();
+    }
+
+    /**
+     * Asynchronously saves a Token (as a {@link String} to the {@link SharedPreferences} file.
+     *
+     * @param key   The name (key) of the Token to save.
+     * @param value The Token's value (as a {@link String}).
+     */
+    public final void saveTokenAsync(
```

Comment:
Not sure we want to call this async... get that "apply" is async and that it could fail.... however there is no way to wire up to respond to an asynchronous response or a callback.  It's fire and forget.... 

I don't know in what scenarios SharedPreference commit would fail... so I think we may want to investigate this.  I believe that apply was introduced as performance optimization for Android... relative to SharePreferences... so I think we should have an opinion for our peers on which to use... providing both is likely to mean that our peers will call both.... I believe that MSAL currently uses apply  exclusively.

Reply:
[This SO thread](https://stackoverflow.com/questions/30475362/when-android-sharedpreferences-commit-returns-false) hints that failure might occur in race-condition cases. While there's no citation or empirical evidence provided, that rationale makes sense to me.

I'm fine with `apply()` - I'll do exclusively that and deal with any complaints later 😁